Create a new page for How Novu works and its content (#822)
novuhq/docs: content/docs/platform/concepts/how-novu-works.mdx

Reviewer: @Aviatorscode2 can we move it to the previous position out of the concept and to be on the root?

Author: Yes, we can.

---

> title: 'How Novu Works'

Author: Yes, the title of each page is all uppercase

> description: 'A high-level overview of Novu’s architecture: Workflows, Triggers, Subscribers, and Environments, and how they connect to support In-app and external channels.'

Reviewer 2: Why are the words here capitalized? Seems like unnecessary capitalization. In the page, these words are often not capitalized.

Author: These are the core concepts. That was why I capitalized them.

Reviewer 2: I would not in this case.

---

import { Step, Steps } from '@/components/steps';

Novu is a notification infrastructure built around the concept of workflows, environments, and subscriber-based delivery. It centralizes multi-channel messaging and supports real-time in-app experiences through a customizable <Method href="/platform/inbox/react/components/inbox">{`<Inbox />`}</Method> component.

> At its core, Novu uses workflows to orchestrate how notifications are sent, step by step, to different channels like Email, SMS, In-App, Push, and Chat. Each workflow is triggered by your application and executed in the context of a specific environment.

Reviewer 2: Why are the different channels capitalized? Seems unnecessary.

## The Inbox component in the notification lifecycle

> The <Method href="/platform/inbox/react/components/inbox">{`<Inbox />`}</Method> component is Novu’s in-app delivery endpoint. It makes it possible for users to receive and manage notifications directly within your application interface.

Reviewer 2: Unclear antecedent. What makes it possible for users to do stuff? Is it the component? I would suggest "This component makes it possible..." 
Or
"This component allows users to..."

Author: Okay

Unlike external channels that rely on third-party providers, the Inbox is internal to your app. It uses the environment’s `application identifier` to establish a secure connection, scoped to your subscribers.

> When a workflow includes an In-App step, messages are routed to the subscriber’s inbox in real time. The Inbox component handles rendering, notification management, users' preferences, and message actions without requiring additional backend logic.

Reviewer 2: ```suggestion
When a workflow includes an in-app step, messages are routed to the subscriber’s inbox in real time. The Inbox component handles rendering, notification management, users' preferences, and message actions without requiring additional backend logic.
```

## Core concepts overview

At a high level, Novu’s architecture is built around modular but connected concepts. Each concept serves a specific role in the delivery pipeline and is scoped to an environment.

### Organization and Environments

Everything starts with your organization, the top-level entity in Novu. Each organization can have multiple environments, such as:
- Development for testing
- Production for live apps
- Any custom environments you need (for example, Staging)

Each environment contains:
- Independent workflows
- Environment-specific subscribers
- Topics
- Separate integrations and credentials
- API keys for secure access (application identifier, secret keys)

This isolation allows teams to test workflows and notifications on one environment without affecting the other environments.

> ### Workflows and Triggers

Reviewer 2: ```suggestion
### Workflows and Triggers

```

> A workflow defines the steps Novu takes to deliver a notification. Steps are either channel steps (Email, In-App, Push, Chat, and SMS) or action steps (Delay and Digest).

Reviewer 2: I am guessing from the context that the capitalized items are capitalized to indicate they are in the UI? The channels seem like they could be generic, but "Delay and Digest" do not. It is very confusing when words are capitalized sometimes, but other times are not.

Workflows are environment-specific and versioned. You trigger them using the REST API by sending an event along with the relevant secret key from the current environment, `subscriberId`, `workflowId`, and `payload`.

Each trigger becomes an event that flows through the workflow. Steps in the workflow generate messages, which are delivered via their corresponding channels or routed to the inbox.

> [Learn more](/platform/concepts/workflows)

Reviewer 2: This link text is vague and uninformative. As a user, I have no idea where I am going if I click it. I usually write something like:

To learn more about workflows, refer to Workflows (or whatever the title of the page is that the link points to).

> ### Subscribers and Topics

Reviewer 2: ```suggestion
### Subscribers and topics
```

A subscriber represents a user in a given environment. Each subscriber has:

- Channel identifiers: Such as email address or phone number.
- A unique `subscriberId`: Used when triggering workflows and resolving the recipient.
- Personal details: You can pass `firstName` and `lastName` when creating the subscriber.
- Custom metadata: Stored under a `data` field, which is a flexible JSON object where you can store any additional user-specific information (for example, plan level, preferences, timestamps).
- Environment-specific preferences: Like preferred channels and opt-in settings.

To send notifications to multiple users, you can group subscribers into a topic. A workflow can be triggered for an individual subscriber or for all subscribers within a topic.

Subscriber resolution is part of workflow execution. Novu uses the subscriber's metadata to determine how messages are personalized, delivered, and stored.

> [Learn more](/platform/concepts/subscribers)

Reviewer 2: See earlier comments about link text.

### Channels and the Inbox component

Each channel represents a delivery medium (Email, SMS, Push, Chat, or In-App). Channels are linked to provider configurations known as integrations, which are defined per environment.

> In-App messages are delivered to the <Method href="/platform/inbox/react/components/inbox">{`<Inbox />`}</Method> component, which lives in your frontend and listens for messages scoped to the current environment and subscriber.

Reviewer 2: Can you find a less personifying verb than "lives"?

> [Learn more](/platform/integrations/overview)

Reviewer 2: See earlier comments about link text.

### Integrations

Integrations connect Novu to external delivery providers (e.g., SendGrid, Twilio, Firebase Cloud Messaging, Slack). They are configured per environment and mapped to specific channels.

During workflow execution, Novu uses these integrations to send messages for steps that require external delivery. The presence of an integration for a given channel is required for a workflow step to succeed.

> [Learn more]( link here)

Reviewer 2: See earlier comments about link text. Also, link is missing.

> 

Reviewer 2: ```suggestion
```

> ## How it all connects

Reviewer 2: ```suggestion
## How it all connects

```

  </Step>

  <Step title="The workflow steps execute sequentially">
   Each step is evaluated in order. These steps include:
   - Channel steps: deliver a message via Email, SMS, Push, In-App, or Chat.
   - Action steps: introduce logic such as delays or digest aggregation.
  </Step>

  <Step>
  If a step fails (e.g., due to a provider error), Novu retries it without re-executing previous steps. This ensures fault tolerance and message-level traceability.
  </Step>

  <Step title="Messages are delivered">
  Once a step completes, Novu uses the environment’s configured integrations to deliver the message. For example:
  - An Email step uses a provider like SendGrid.
  - An In-App step sends the message to the <Method href="/platform/inbox/react/components/inbox">{`<Inbox />`}</Method> component in your frontend.
  - An SMS step might use Twilio.
  </Step>

  <Step title="Logs and metadata are recorded">
  Each triggered workflow becomes an event with a unique `transactionId`. Novu records:
  - Which steps ran.
  - Whether each step succeeded or failed.
  - How long each step took.
  - What messages were generated and sent.
  </Step>

</Steps>

> This pipeline is designed to be consistent, observable, and scalable across channels and environments.

Reviewer 2: ```suggestion
The Novu Inboc pipeline is designed to be consistent, observable, and scalable across channels and environments.
```
